crates/mun_runtime/src/static_type_map.rs

```diff
@@ -1,15 +1,16 @@
+use parking_lot::ReentrantMutex;
 use std::any::TypeId;
+use std::cell::RefCell;
 use std::collections::HashMap;
-use std::sync::RwLock;
 
 pub struct StaticTypeMap<T: 'static> {
-    map: RwLock<HashMap<TypeId, &'static T>>,
+    map: ReentrantMutex<RefCell<HashMap<TypeId, &'static T>>>,
```

Comment:
```
ReentrantMutexGuard does not give mutable references to the locked data. Use a RefCell if you need this.
```
This is because you could then get multiple `mut` references when recursively locking the mutex.